fil_fungible_token/src/receiver/mod.rs

```diff
@@ -1 +1,56 @@
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::{address::Address, econ::TokenAmount, error::ExitCode};
+use num_traits::Zero;
+use types::TokensReceivedParams;
+
+use crate::runtime::messaging::{Messaging, RECEIVER_HOOK_METHOD_NUM};
+use crate::token::TokenError;
+
 pub mod types;
+
+#[derive(Debug)]
+pub struct ReceiverHookGuard<T> {
```

Comment:
I would just call this `ReceiverHook`, or maybe `ReceiverHookThunk`, tho that might be too much jargon. Please add documentation comments explaining its behaviour.